exporter/exporterhelper/common.go

```diff
@@ -181,6 +176,15 @@ func newBaseExporter(set exporter.CreateSettings, signal component.DataType, req
 	}
 	be.connectSenders()
 
+	// If retry sender is disabled then disable requeuing in the queue sender.
+	// TODO: Make re-enqueuing configurable on queue sender instead of relying on retry sender.
+	if qs, ok := be.queueSender.(*queueSender); ok {
+		// if it's not retrySender, then it is disabled.
+		if _, ok = be.retrySender.(*retrySender); !ok {
+			qs.requeuingEnabled = false
+		}
+	}
+
```

Comment:
I am not sure that this was intentional or a bug of the previous implementation. Happy to keep it as it is now, but my bet is that it was just a bug in the previous implementation.